public/app/features/browse-dashboards/components/SearchView.tsx

```diff
@@ -93,25 +96,25 @@ export function SearchView({
     [selectionChecker, dispatch]
   );
 
-  if (value.totalRows === 0) {
-    return (
-      <div style={{ width }}>
-        <EmptyState
-          button={
-            <Button variant="secondary" onClick={stateManager.onClearSearchAndFilters}>
-              <Trans i18nKey="browse-dashboards.no-results.clear">Clear search and filters</Trans>
-            </Button>
-          }
-          message={t('browse-dashboards.no-results.text', 'No results found for your query')}
-          variant="not-found"
-          role="alert"
-        />
-      </div>
+  if (value?.totalRows === 0) {
```

Comment:
I think this logic doesn't work because `initialLoadingView` has 50 empty rows by default, so it's not caught by this check and the UI shows the `No data` message.